README.md

```diff
@@ -11,42 +11,54 @@
-- `lineintegral`, `surfaceintegral`, and `volumeintegral` are available as aliases for `integral` that first verify that `geometry` has the appropriate number of parametric dimensions
+These solvers have support for integrand functions that produce scalars, vectors, and **Unitful.jl** `Quantity` types. While HCubature.jl does not natively support `Quantity` type integrands, this package provides a compatibility layer to enable this feature.
 
-# Example Usage
+## Usage
+
+```julia
+integral(f, geometry)
+```
+Performs a numerical integration of some integrand function `f(p::Meshes.Point)` over the domain specified by `geometry`. A default integration method will be automatically selected according to the geometry: `GaussKronrod` for 1D, and `HAdaptiveCubature()` for all others.
+
+```julia
+integral(f, geometry, algorithm, FP=Float64)
+```
+Performs a numerical integration of some integrand function `f(p::Meshes.Point)` over the domain specified by `geometry` using the specified integration algorithm, e.g. `GaussKronrod()`.
+
+Optionally, a fourth argument can be provided to specify the floating point precision level desired. This setting can be manipulated if your integrand function produces outputs with alternate floating point precision (e.g. `Float16`, `BigFloat`, etc) AND you'd prefer to avoid implicit type promotions.
+
+```julia
+lineintegral(f, geometry)
+surfaceintegral(f, geometry)
+volumeintegral(f, geometry)
+```
+Alias functions are provided for convenience. These are simply wrappers for `integral` that first validate that the provided `geometry` has the expected number of parametric/manifold dimensions. Like with `integral` in the examples above, the `algorithm` can also be specified as a third-argument.
+- `lineintegral` for curve-like geometries or polytopes (e.g. `Segment`, `Ray`, `BezierCurve`, `Rope`, etc)
+- `surfaceintegral` for surfaces (e.g. `Disk`, `Sphere`, `CylinderSurface`, etc)
+- `volumeintegral` for (3D) volumes (e.g. `Ball`, `Cone`, `Torus`, etc)
+
+# Demo
 
 ```julia
 using Meshes
 using MeshIntegrals
 
-# Define a unit circle on the xy-plane
-origin = Point(0,0,0)
-ẑ = Vec(0,0,1)
-xy_plane = Plane(origin,ẑ)
-unit_circle_xy = Circle(xy_plane, 1.0)
-
-# Approximate unit_circle_xy with a high-order Bezier curve
-unit_circle_bz = BezierCurve(
-    [Point(cos(t), sin(t), 0.0) for t in range(0,2pi,length=361)]
+# Define a path that approximates a sine-wave on the xy-plane
+mypath = BezierCurve(
+    [Point(t*u"m", sin(t)*u"m", 0.0u"m") for t in range(-pi, pi, length=361)]
 )
 
-# A Real-valued function
-f(x, y, z) = abs(x + y)
-f(p) = f(to(p)...)
+# Map f(::Point) -> f(x, y, z)
+f(p::Meshes.Point) = f(to(p)...)
 
-integral(f, unit_circle_xy, GaussKronrod())
-    # 0.000170 seconds (5.00 k allocations: 213.531 KiB)
-    # ans == 5.656854249525293 m^2
+# Integrand function in units of Volts/meter
+f(x, y, z) = exp(x^2 + y^2) * u"V/m"
```

Comment:
```suggestion
f(x, y, z) = exp(x^2 + y^2) * u"V/m"
```
Have you tried that this actually works? I had something similar a while back, which didn't work because of the `exp`. Since `x` and `y` have units, I think `exp` throws an error because exponentiating a unit is usually not meaningful.

Reply:
Good catch! I’m writing these updates on the go, so haven’t actually tried running them yet. These coordinates are definitely Unitful so won’t work directly inside of `exp`.

Reply:
You could do something like `f(x, y, z) = exp(u"1/m^2" * (x^2 + y^2)) * u"V/m"`, but I would be interested whether there is another (maybe nicer solution). In my case, `exp` is defined in a function of another package (well, my package...), so it's not so easy to apply the unit within the `exp`.

Reply:
I see you simply use `ustrip` there, which of course also works.

Reply:
I want to emphasize simplicity and capability, just need to find the right balance. Maybe there’s a cleaner way to modify the wrapper `f(p)` to strip the units there?

Reply:
I just commit a possible fix for this issue. I’ll hopefully be back on later this evening to run the code and see if it actually works.

Reply:
I just ran it and it didn't work. I proposed a solution below, which worked for me.